packages/EasyCore/src/Doctrine/Type/DateTimeMicrosecondsType.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyCore\Doctrine\Type;
+
+use DateTime;
+use DateTimeInterface;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\ConversionException;
+use Doctrine\DBAL\Types\Type;
+
+final class DateTimeMicrosecondsType extends Type
+{
+    private const TYPENAME = 'datetime';
+
+    /**
+     * @param mixed $value
+     *
+     * @throws \Doctrine\DBAL\Types\ConversionException
+     */
+    public function convertToDatabaseValue($value, AbstractPlatform $platform): ?string
+    {
+        if ($value === null) {
+            return $value;
+        }
+
+        if ($value instanceof DateTimeInterface) {
+            return $value->format('Y-m-d H:i:s.u');
+        }
+
+        throw ConversionException::conversionFailedInvalidType(
+            $value,
+            $this->getName(),
+            ['null', 'DateTime']
+        );
+    }
+
+    /**
+     * @param mixed $value
+     *
+     * @throws \Doctrine\DBAL\Types\ConversionException
+     */
+    public function convertToPHPValue($value, AbstractPlatform $platform): ?DateTimeInterface
+    {
+        if ($value === null || $value instanceof DateTimeInterface) {
+            return $value;
+        }
+
+        $val = DateTime::createFromFormat('Y-m-d H:i:s.u', $value) ?: \date_create($value);
+
+        if ($val !== false) {
+            return $val;
+        }
+
+        throw ConversionException::conversionFailedFormat(
+            $value,
+            $this->getName(),
+            'Y-m-d H:i:s.u'
+        );
+    }
+
+    public function getName(): string
+    {
+        return self::TYPENAME;
+    }
+
+    public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform): string
+    {
+        if (isset($fieldDeclaration['version']) && $fieldDeclaration['version'] === true) {
+            return 'TIMESTAMP';
+        }
+
+        return 'DATETIME(6)';
```

Comment:
Let's move this to a `FORMAT_DB_DATETIME` const.

Reply:
Done.